internal/colerrs/README.md

```diff
@@ -0,0 +1,17 @@
+# Internal collector errors
+
+This package includes helpers propagating errors from exporters or processors
```

Comment:
The behaviour around error propagation for synchronous pipelines should be documented in a place that's more visible than the package's readme :)